picard/coverart/providers/caa.py

```diff
@@ -122,30 +304,79 @@ def __init__(self, parent=None, types=None):
         self.buttonbox.rejected.connect(self.reject)
         self.buttonbox.helpRequested.connect(self.help)
 
+        self.fix_buttons()
+
+    def fill_lists(self, includes, excludes):
+        """Fill dialog listboxes.
+
+        First clears the contents of the three listboxes, and then populates the listboxes
+        from the dictionary of translated standard CAA types, using the provided 'includes'
+        and 'excludes' lists to determine the appropriate list for each type.
+
+        Arguments:
+            includes -- list of standard image types to place in the "Include" listbox
+            excludes -- list of standard image types to place in the "Exclude" listbox
+        """
+        self.list_include.clear()
+        self.list_exclude.clear()
+        self.list_ignore.clear()
+        for title in self.image_types_by_display_title:
+            name = self.image_types_by_display_title[title]
+            if name in includes:
+                self.list_include.addItem(title)
+            elif name in excludes:
+                self.list_exclude.addItem(title)
+            else:
+                self.list_ignore.addItem(title)
+
     def help(self):
         webbrowser2.goto('doc_cover_art_types')
 
-    def uncheckall(self):
-        self._set_checked_all(False)
+    def get_selected_types_include(self):
+        return [self.image_types_by_display_title[self.list_include.item(index).text()] for index in range(self.list_include.count())] or ['front']
 
-    def checkall(self):
-        self._set_checked_all(True)
+    def get_selected_types_exclude(self):
+        return [self.image_types_by_display_title[self.list_exclude.item(index).text()] for index in range(self.list_exclude.count())] or ['none']
 
-    def _set_checked_all(self, value):
-        for item in self._items.keys():
-            item.setChecked(value)
+    def clear_focus(self, lists):
+        for temp_list in lists:
+            temp_list.clearSelection()
+        self.fix_buttons()
 
-    def get_selected_types(self):
-        return [typ['name'] for item, typ in self._items.items() if
-                item.isChecked()] or ['front']
+    def reset_to_defaults(self):
+        self.fill_lists(_CAA_IMAGE_TYPE_DEFAULT_INCLUDE, _CAA_IMAGE_TYPE_DEFAULT_EXCLUDE)
+        self.fix_buttons()
+
+    def fix_buttons(self):
+        has_items_include = self.list_include.count()
+        has_items_exclude = self.list_exclude.count()
+        has_items_ignore = self.list_ignore.count()
+
+        has_selected_include = True if self.list_include.selectedItems() else False
```

Comment:
I tried that before without the bool() and got the following error:

    Traceback (most recent call last):
      File ".\picard\coverart\providers\caa.py", line 443, in select_caa_types
        self, config.setting["caa_image_types"], config.setting["caa_image_types_to_omit"])
      File ".\picard\coverart\providers\caa.py", line 380, in run
        dialog = CAATypesSelectorDialog(parent, types_include, types_exclude)
      File ".\picard\coverart\providers\caa.py", line 307, in __init__
        self.fix_buttons()
      File ".\picard\coverart\providers\caa.py", line 363, in fix_buttons
        self.arrows_include.button_add.setEnabled(has_items_ignore and has_selected_ignore)
    TypeError: setEnabled(self, bool): argument 1 has unexpected type 'list'

It seems that the error occurs because it's trying to AND a bool with a list.  That's why I went with forcing it to a True or False directly.

It works fine with the bool(), so I'll make that change.